packages/runner/src/builtins/compile-and-run.ts

```diff
@@ -191,40 +326,80 @@ export function compileAndRun(
     // Capture requestId for this compilation run
     const thisRequestId = requestId;
 
-    const compilePromise = runtime.harness.run(program)
```

Comment:
There is already a recipe-manager in runner that caches compiled ones per program hash. So maybe directly calling runtime.harness.run bypasses that?

The recipe manager code is a little messy. So it would be worth understanding the flows there and around runtime.harness.run and clean it up, making sure this path hits the cache.

Let's not add another caching layer here that only compileAndRun uses.